dsl/response.go

```diff
@@ -0,0 +1,194 @@
+package dsl
+
+import (
+	"goa.design/goa/eval"
+	"goa.design/goa/expr"
+)
+
+// Response describes a HTTP or a gRPC response. Response describes both success
+// and error responses. When describing an error response the first argument is
+// the name of the error.
+//
+// While a service method may only define a single result type, Response may
+// appear multiple times to define multiple success HTTP responses. In this case
+// the Tag expression makes it possible to identify a result type attribute and
+// a corresponding string value used to select the proper success response (each
+// success response is associated with a different tag value). gRPC responses
+// may only define one success response.
+//
+// Response may appear in an API or service expression to define error responses
+// common to all the API or service methods. Response may also appear in a
+// method expression to define both success and error responses specific to the
+// method.
+//
+// Response accepts one to three arguments. Success response accepts a status
+// code as first argument. If the first argument is a status code then a
+// function may be given as the second argument. This function may provide a
+// description and describes how to map the result type attributes to transport
+// specific constructs (e.g. HTTP headers and body, gRPC metadata and message).
+//
+// The valid invocations for successful response are thus:
+//
+// * Response(status)
+//
+// * Response(func)
+//
+// * Response(status, func)
+//
+// Error responses additionally accept the name of the error as first argument.
+//
+// * Response(error_name, status)
+//
+// * Response(error_name, func)
+//
+// * Response(error_name, status, func)
+//
+// By default (i.e. if Response only defines a status code) then:
+//
+//    - success HTTP responses use code 200 (OK) and error HTTP responses use code 400 (BadRequest)
+//    - success gRPC responses use code 0 (OK) and error gRPC response use code 2 (Unknown)
+//    - The result type attributes are all mapped to the HTTP response body or gRPC response message.
+//
+// Example:
+//
+//    Method("create", func() {
+//        Payload(CreatePayload)
+//        Result(CreateResult)
+//        Error("an_error")
+//
+//        HTTP(func() {
```

Comment:
Would be cool to add a `GRPC` call as well.

Reply:
I thought I would add it along with the gRPC PR.